client/client.go

```diff
@@ -316,11 +317,83 @@ func makeQueryString(data interface{}) (string, error) {
 	return "", fmt.Errorf("unsupported query string data: %#v", data)
 }
 
+// Remove all custom request serializer logic once APP-1331 is rolled out.
+type serializer func(interface{}) ([]byte, error)
+
+func serializeUnityCatalogDeleteRequest(data interface{}) ([]byte, error) {
+	m := make(map[string]interface{})
+	// Reflectively look up the Force field of data, and if it is true, set the
+	// "force" field of m to true.
+	setField := func(v reflect.Value, i int, structName, bodyName string) {
+		if v.Type().Field(i).Name == structName {
+			// check all available kinds and set m[bodyName] to the value of the field
+			switch v.Field(i).Kind() {
+			case reflect.Bool:
+				m[bodyName] = v.Field(i).Bool()
+			case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+				m[bodyName] = v.Field(i).Int()
+			case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+				m[bodyName] = v.Field(i).Uint()
+			case reflect.Float32, reflect.Float64:
+				m[bodyName] = v.Field(i).Float()
+			case reflect.String:
+				m[bodyName] = v.Field(i).String()
+			}
+		}
+	}
+	v := reflect.ValueOf(data)
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+	if v.Kind() == reflect.Struct {
+		for i := 0; i < v.NumField(); i++ {
+			setField(v, i, "Force", "force")
+			setField(v, i, "MetastoreId", "metastore_id")
+		}
+	}
+
+	return json.MarshalIndent(m, "", "  ")
+}
+
+var metastorePath = regexp.MustCompile("/api/2.1/unity-catalog/metastores/[^/]+")
```

Comment:
Does TF use 2.1 for UC? Don’t remember from top of my head

Reply:
I copied these from the autogenerated code, so I think they should be right. I think this should be covered by the integration test cases?

Reply:
TF does not fully rely on autogenerated code yet. that's why we have Go SDK migration ;)